[ANDROID-11430] Add uses-feature checks

jeprubio opened the pull request:
### :tickets: Jira ticket
[ANDROID-11430](https://jira.tid.es/browse/ANDROID-11430)

### :goal_net: What's the goal?
We need a check for the 'uses-feature' permissions as well as the current ones.
Uses-feature nodes have an optional "required" attribute that we also want to check it has not changed.

### :construction: How do we do it?
Adding code in order to be able to serialize and read the features from the manifest and also compare their values.

### To execute this:
There are instructions on how to use it with the sample app it's in the repo on the README.md but mostly they are:
> Uncomment the highlighted lines in the settings.gradle.kts and the build.gradle.kts files to include the sample project.
> ./gradlew jar
> ./gradlew publishToMavenLocal
> ./gradlew checkPermissions

pmartinbTEF reviewed (changes requested):
I added a feature to the sample app manifest but it passed anyway:
    <uses-feature android:name="android.hardware.bluetooth" android:required="true" />
Adding a new permission works as expected

jeprubio commented:
> I added a feature to the sample app manifest but it passed anyway: Adding a new permission works as expected

Could you add more info? which permission has it let you passed?

> I added a feature to the sample app manifest but it passed anyway: Adding a new permission works as expected

Could it be one of these permissions with protectionLevel? Like the following:

```
<permission
	android:name="${applicationId}.permission.PackageReplacedReceiverPermission"
	android:protectionLevel="signature"
	/>
```
I've checked and you are right, this fails in here as well as in our app, that permission is from our app and if I remove it there the plugin doesn't complain at all, it's kind of ignored, I'll check.

pmartinbTEF reviewed (approved):
Now it works. Good job!

dpastor reviewed (approved):
👍
